tools/ClangTidyHash.sh

```diff
@@ -3,16 +3,22 @@
 # Distributed under the MIT License.
 # See LICENSE.txt for details.
 
-if [ $# != 3 ]; then
-    echo "Wrong number of arguments passed to ClangTidyHash.sh"
+if [ $# != 3 ] && [ $# != 4 ]; then
+    echo "Wrong number of arguments passed to ClangTidyHash.sh, $#"
     echo "Expecting BUILD_DIR SOURCE_DIR HASH"
+    echo "or BUILD_DIR SOURCE_DIR HASH NUM_THREADS_TO_RUN_ON"
     exit 1
 fi
 
-BUILD_DIR=$1
-SOURCE_DIR=$2
+BUILD_DIR=$(realpath $1)
+SOURCE_DIR=$(realpath $2)
```

Comment:
I missed this before, but I think realpath has portability issues.  Someone on OSX should verify that this works.  @kidder?